Automattic/wp-calypso
# Store: Attempt to automatically fix a site with required plugins disabled

Fixes #20918 – Loading a store with woocommerce disabled puts you in a strange error state. This PR attempts to detect whether [the required plugins](https://github.com/Automattic/wp-calypso/blob/4a92ea0e203f957e55f5ba675912b230f20f1483/client/extensions/woocommerce/lib/get-required-plugins.js) are disabled, and if they are, it drops the user into the `RequiredPluginsInstallView`, which runs through the install process to fix the issues. This fixes anything from "woocommerce is inactive" to "all plugins are deleted".

This also adds a "title" prop to `RequiredPluginsInstallView`, so that we can customize the message here. I've used "Updating your store", but we could possibly do "Fixing your store" (fits with the builder illustration).

<img width="319" alt="screen shot 2018-03-30 at 11 40 50 am" src="https://user-images.githubusercontent.com/541093/38144746-2d6b0b02-3414-11e8-8b35-1c78b736a974.png">

I've based this on the counts PR, #23602, since it cleans up some of the API requests in the sidebar, and I've also added code to prevent data fetches in the sidebar until we know we have a "valid" site.

**To test**

- Deactivate some plugins on your site
- Try to load up any store page
- You should get the above screen, then once it's finished, it will load the page you requested

I'm attempting to test the signup flow, but the transfer is taking a long time - so if someone else could try that would be great :)
